src/Layer/VNode.cpp

```diff
@@ -32,10 +33,23 @@ void VNode::observe(const VNodePtr& sender)
   {
     sender->m_observers.push_back(shared_from_this());
   }
+#else
+  if (auto it = std::find_if(
+        sender->m_observers.begin(),
+        sender->m_observers.end(),
+        [&](const auto& o) { return o.lock() == this; });
+      it == sender->m_observers.end())
+  {
+    this->ref();
```

Comment:
1. Why we still need `VNode::observe` if we use VGG`s smart pointer ?
2. What's the point to call `ref()` if we are not using shared_ptr?